src/AutoRest.CSharp.V3/Output/Models/Types/ObjectType.cs

```diff
@@ -101,7 +101,12 @@ private IEnumerable<ObjectTypeConstructor> BuildConstructors()
             }
 
             yield return new ObjectTypeConstructor(
-                BuilderHelpers.CreateMemberDeclaration(Type.Name, Type, "internal", _sourceTypeMapping?.DefaultConstructor),
+                BuilderHelpers.CreateMemberDeclaration(
+                    Type.Name,
+                    Type,
+                    // inputs have public ctor by default
+                    _objectSchema.IsInput ? "public" : "internal",
+                    _sourceTypeMapping?.DefaultConstructor),
```

Comment:
I was wondering about them both being internal. I just assumed this initial implementation didn't matter or not. Something that would have gotten fixed up when they actually get called.

Reply:
Don't want to break existing generated clients.